pkg/ccl/backupccl/testdata/backup-restore/online-restore-in-progress-imports

```diff
@@ -0,0 +1,83 @@
+# This test ensures that online restore skips restoring tables 
+# undergoing an in progress import
+
+reset test-nodelocal
+----
+
+new-cluster name=s1 disable-tenant
+----
+
+
+exec-sql
+CREATE DATABASE d;
+USE d;
+CREATE TABLE foo (i INT PRIMARY KEY, s STRING);
+CREATE TABLE foofoo (i INT PRIMARY KEY, s STRING);
```

Comment:
Do we need to think about foreign keys or views? Is is possible to start an IMPORT into a table that other tables depend on? or that depends on other tables? I'm wondering if a little `continue` will severe links?

... or were we handling that already, before the `eligible` check?

Reply:
Yeah, i have no idea. We could run a schema only restore on these tables instead, but again, that might sever the dependency links

Reply:
can you just run this locally with some FKs added in to these lines and see what happens? I don't know if we let you take the table into IMPORTING while it is still in the dep graph, but if we don't we can just stop thinking about what restore needs to do since it doesn't need to worry about it.

Reply:
looks like we're ok. changed the schema for an importing table to:
```
CREATE TABLE baz (i INT PRIMARY KEY, s STRING);
INSERT INTO baz VALUES (1, 'x'),(2,'y'),(3,'z');
CREATE TABLE foo (
  i INT PRIMARY KEY,
  baz_1 INT NOT NULL REFERENCES baz (i),
   s STRING);
```
and the import failed with
```
 pq: all non-target columns in IMPORT INTO must be nullable or have default expressions, or have computed expressions but violated by column "baz_1"
```

Seems like import's known limitations should clarify this:
https://www.cockroachlabs.com/docs/v23.2/import-into#known-limitations

Reply:
and if qux REFERENCES foo are we good too?

Reply:
rats, nope. Import on foo passes with this schema:
```
CREATE TABLE foo (
  i INT PRIMARY KEY,
   s STRING);
CREATE VIEW foo_view (counter) AS SELECT count(*) FROM foo;
```

Reply:
Aaand and import into `foofoo` also passes 😞 
```
CREATE TABLE foofoo (i INT PRIMARY KEY, s STRING);
INSERT INTO foofoo VALUES (10, 'x0');
CREATE TABLE foo (
  i INT PRIMARY KEY,
  foofoo_1 INT REFERENCES foofoo (i),
   s STRING);
----
```

Reply:
so what happens when we restore the backup where we clip `foo` out of the graph? is foo_view still there? does foofoo have broken backrefs?

Reply:
yeah, I've got to run a few experiments here. will report back.

Reply:
I added some toy test cases in the [in-progress-restore-exp](https://github.com/cockroachdb/cockroach/pull/119700/commits/e700186a1181d6f63bac46d7ce4a4071a92e74dd#diff-be6c128a865ce6152e4eed83763bdec3b5db19aea731b4b2c46a2a0a177b3cff) test file, to understand how conventional restore restores a table with an in-progress import  that other tables depend on. Conventional restore works well!

But I also learned that we can't just clip the importing table from the graph-- the restore will fail. So this approach does not seem like a good idea.

As an alternative approach, what if we conducted a schema_only restore on the importing tables? The restore should succeed (I didn't test this); however, existing foreign key constraints that depend on the table will be violated. For example, consider the following setup:
```
exec-sql
CREATE DATABASE d;
USE d;
CREATE TABLE foofoo (i INT PRIMARY KEY, s STRING);
INSERT INTO foofoo VALUES (10, 'x0');
CREATE TABLE foo (
  i INT PRIMARY KEY,
  foofoo_1 INT REFERENCES foofoo (i),
   s STRING);
INSERT INTO foo VALUES (3,10,'hey')
----

import expect-pausepoint tag=aa
IMPORT INTO foofoo (i,s) CSV DATA ('nodelocal://1/export1/export*-n*.0.csv');
----
job paused at pausepoint

exec-sql
BACKUP INTO 'nodelocal://1/cluster/';
----
```

If RESTORE runs a schema only restore on foofoo, then the foreign key on table foo will be violated.